packages/next-on-pages/src/utils/fs.ts

```diff
@@ -125,3 +127,54 @@ export async function copyFileWithDir(sourceFile: string, destFile: string) {
 	await mkdir(dirname(destFile), { recursive: true });
 	await copyFile(sourceFile, destFile);
 }
+
+/**
+ * Reads all directories in a directory.
+ *
+ * @param path Path to read directories from.
+ * @returns Array of all directories in a directory.
+ */
+export async function readDirectories(
+	basePath: string,
+): Promise<DirectoryInfo[]> {
+	try {
+		const files = await readdir(basePath, { withFileTypes: true });
+
+		const dirs = await Promise.all(
+			files.map(async file => {
+				const path = normalizePath(join(basePath, file.name));
+				const isSymbolicLink = file.isSymbolicLink();
+				const isDirectory =
+					file.isDirectory() || (isSymbolicLink && (await validateDir(path)));
```

Comment:
i wanted to avoid having to do the `stat()` for the file in the fs (which `validateDir` uses) if it wasn't a symbolic link. it's unnecessary extra work when it's not a symlink.

Reply:
mh... ok, I'm fine with it, unfortunately it makes this line less readable, but it's not the end of the world (especially if we get some performance improvements 👍)